tests/utils.go

```diff
@@ -108,15 +107,39 @@ func IsPVCDeletedEventually(pvcName string) bool {
 
 // VerifyStorageClassParams verifies the volume properties set at creation time
 func VerifyStorageClassParams(parameters map[string]string) {
+	vol, err := ZFSClient.WithNamespace(OpenEBSNamespace).
+		Get(pvcObj.Spec.VolumeName, metav1.GetOptions{})
+	gomega.Expect(err).To(gomega.BeNil(), "while fetching the zfs volume {%s}", vol.Name)
+	// Check for file system type
+	if parameters["fstype"] == "zfs" {
+		parameters["type"] = "filesystem"
+	} else {
+		parameters["type"] = "volume"
+	}
+	generateThinProvisionParams(parameters)
+	delete(parameters, "fstype")
+
 	for propertyKey, propertyVal := range parameters {
-		if propertyKey != "fstype" && propertyKey != "thinprovision" {
-			vol, err := ZFSClient.WithNamespace(OpenEBSNamespace).
-				Get(pvcObj.Spec.VolumeName, metav1.GetOptions{})
-			gomega.Expect(err).To(gomega.BeNil(), "while fetching the zfs volume {%s}", vol.Name)
-			status := IsPropUpdatedEventually(vol, propertyKey, propertyVal)
-			gomega.Expect(status).To(gomega.Equal(true), "while updating {%s%}={%s%} {%s}", propertyKey, propertyVal, vol.Name)
+		status := IsPropUpdatedEventually(vol, propertyKey, propertyVal)
+		gomega.Expect(status).To(gomega.Equal(true), "while updating {%s%}={%s%} {%s}", propertyKey, propertyVal, vol.Name)
+	}
+
+}
+
+// It populates the map for thing provisioning params
+// Refer https://github.com/openebs/zfs-localpv/issues/560#issuecomment-2232535073
+func generateThinProvisionParams(parameters map[string]string) {
+	if parameters["fstype"] == "zfs" {
+		parameters["quota"] = capacity
+		parameters["reservation"] = defaultReservation
+		if parameters["thinprovision"] == "no" {
+			parameters["reservation"] = capacity
 		}
+	} else {
+		parameters["quota"] = "-"
+		parameters["reservation"] = defaultReservation
 	}
+	delete(parameters, "thinprovision")
 }
```

Comment:
We are not supposed to set the parameters manually, it would happen from the code.